Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -159,7 +159,13 @@ public class MemoryBookmarkFolder: BookmarkFolder, SequenceType {
         }
 
         public mutating func next() -> BookmarkNode? {
-            return index < children.count ? children[index++] : nil
+            if index < children.count {
```

Comment:
> Yeah I think you are right. I think this should be:
> 
> if index < children.count {
>     index += 1
>     return children[index]
> }  else ...

I don't think that's right -- the `++` postfix operator means "access, then increment". I think it's correct how it is now: get the result at the current index, then increment the index, then return the result.

Alternatively, you could use `defer` as suggested [here](http://ericasadun.com/2016/01/26/welcoming-our-new-swift-2-2-overlords/):
```
defer { index += 1 }
return children[index]
```